apps/minifront/src/state/staking/index.ts

```diff
@@ -311,21 +306,32 @@ export const createStakingSlice = (): SliceCreator<StakingSlice> => (set, get) =
   loadAndReduceBalances: async () => {
     const balancesByAccount = await getBalancesByAccount();
 
+    const stakingTokenMetadata = await getStakingTokenMetadata();
+
     // It's slightly inefficient to reduce over an array twice, rather than
     // combining the reducers into one. But this is much more readable; and
     // anyway, `balancesByAccount` will be a single-item array for the vast
     // majority of users.
-    const unstakedTokensByAccount = balancesByAccount.reduce(toUnstakedTokensByAccount, new Map());
-    const accountSwitcherFilter = balancesByAccount.reduce(toAccountSwitcherFilter, []);
-
+    const unstakedTokensByAccount = balancesByAccount.reduce(
+      (acc: Map<number, ValueView>, cur: BalancesByAccount) =>
+        toUnstakedTokensByAccount(acc, cur, stakingTokenMetadata),
+      new Map(),
+    );
+    const accountSwitcherFilter = balancesByAccount.reduce(
+      (acc: number[], cur: BalancesByAccount) =>
+        toAccountSwitcherFilter(acc, cur, stakingTokenMetadata),
+      [],
+    );
```

Comment:
Hm, I'd find this more readable if you convert the reducer functions to higher-order functions that accept the metadata — e.g., 
```suggestion
    const unstakedTokensByAccount = balancesByAccount.reduce(
      toUnstakedTokensByAccount(stakingTokenMetadata),
      new Map(),
    );
    const accountSwitcherFilter = balancesByAccount.reduce(
      toAccountSwitcherFilter(stakingTokenMetadata),
      [],
    );
```

Just my preference, though — feel free to ignore.

Reply:
I agree that your option is better, but I can't get it to work. Could you give me an idea of what else I should do to make it work?

Reply:
@Valentine1898 it's not that important, so feel free to ignore this. But to make this change work, you'd also need to update the reducer functions to be higher-order functions (i.e., functions that return other functions). e.g., 

```ts
const toAccountSwitcherFilter = (stakingTokenMetadata: Metadata) => (
  accountSwitcherFilter: number[],
  curr: BalancesByAccount,
) => {
```